### PR TITLE
Kernel IR refactoring: part 4.1

### PR DESCRIPTION
**Pull request by tlemo**
A small followup to the previous iteration, completing a few unfinished parts around kir::Allocate.

### CODE REVIEW

**Review by csarofeen — CHANGES_REQUESTED** [review]
Blocking for now as Indexing is being rebased.

File: torch/csrc/jit/codegen/cuda/ir_iostream.cpp [hunk 1]
```diff
@@ -836,46 +845,24 @@ void IRPrinter::handle(const kir::IfThenElse* ite) {
 
 void IRPrinter::handle(const kir::Allocate* a) {
   indent();
-  if (a->buffer()->getValType().value() == ValType::TensorView) {
-    auto tv = a->buffer()->as<TensorView>();
-
+  if (a->buffer()->getValType().value() == ValType::KirTensorView) {
+    const auto tv = a->buffer()->as<kir::TensorView>();
+    TORCH_INTERNAL_ASSERT(tv->domain()->nDims() > 0);
```
Comment (csarofeen) [inline torch/csrc/jit/codegen/cuda/ir_iostream.cpp:862]: I'm not 100% certain we won't actually send a 0-dim tensor view to this. That would indicate it's just a scalar.

Comment (csarofeen) [reply torch/csrc/jit/codegen/cuda/ir_iostream.cpp:862]: Actually it looks like we were supporting it.

**Review by csarofeen — APPROVED** [review]

File: torch/csrc/jit/codegen/cuda/ir_iostream.cpp (hunk 1 above)
Comment (tlemo) [reply torch/csrc/jit/codegen/cuda/ir_iostream.cpp:862]: Good question. Yes, the code was in there but never used as far as I can tell. 

More importantly, this is a good example where we can consider different trade-offs between Fusion IR and Kernel IR: in the former the benefit of treating 0-dimensional Tensors as scalars makes sense. We can, and should, eliminate this duality of representing scalars during lowering.

**Pull request merged**